### PR TITLE
Add UI views for uploading files

### PR DESCRIPTION
**Pull request by wlad031**
screen shot 2017-02-26 at 17 08 34

### CODE REVIEW

File: ui/views.py [hunk 1]
```diff
@@ -109,6 +108,68 @@ def register():
...
+
+    if request.method == 'POST':
+
+        if not form.validate_on_submit():
+            archived_images_errors = form.archived_images.errors
+            for error in archived_images_errors:
+                flash(error, 'danger')
+            return _render_template('sample_add.html', form=form)
+
+        file = form.archived_images.data
```
Comment (Imperat) [inline ui/views.py:149]: It's not recommended to override standard python function (file).

File: ui/views.py [hunk 2]
```diff
@@ -109,6 +108,68 @@ def register():
...
+        files = {file.name: (file.filename, file)}
+        r = requests.post(api_url + '/neural/sample/add',
```
Comment (Imperat) [inline ui/views.py:151]: For future. You can hold maximum line length such as:
```
requests.post(
    api_url + '/neural/sample/add', headers=headers, files=files)
```
It looks more pretty!

File: ui/views.py [hunk 3]
```diff
@@ -109,6 +108,68 @@ def register():
...
+
+        file = form.image.data
+        files = {file.name: (file.filename, file)}
+        r = requests.post(api_url + '/neural/recognize',
+                          headers=headers,
+                          files=files)
+
+        # Successful upload
+        if r.status_code == 200:
+            flash('You have been successfully uploaded your image', 'info')
```
Comment (dekarti) [inline ui/views.py]: "Your image has been successfully uploaded"
or
"You have successfully uploaded your image"

File: ui/views.py [hunk 4]
```diff
@@ -109,6 +108,68 @@ def register():
...
+            return _render_template('index.html', form=form)
+
+        # Errors
+        flash(json.loads(r.text)['error'], 'danger')
+        return _render_template('index.html', form=form)
+
+    else:
```
Comment (dekarti) [inline ui/views.py]: Does it make sense?

**Comment by wlad031** [comment]
Some problems fixed. But guys, let's code and look for bugs, and not look for ideological and code-style mistakes. We have not enough time for it.

**Review by Imperat — APPROVED** [review]

File: templates/layout.html [hunk 5]
```diff
@@ -15,14 +15,16 @@
...
-                    <a href="/#" class="not-link">{{ user.username }}</a>
-                <li role="presentation"><a href="/logout">Log Out</a>
-                    {% else %}
-                <li role="presentation" class="active"><a href="/login">Sign In</a>
-                <li role="presentation"><a href="/join">Sign Up</a>
-                    {% endif %}
+                {% if user %}
+                        {# if user is admin or manager #}
+                        {% if user.role == 2 or user.role == 1 %}
+                            <li role="presentation"><a href="/sample/add">Add learning sample</a>
```
Comment (dansagunov) [inline templates/layout.html:21]: Don't the `<li>` tag should be closed with `</li>` here and then?

**Comment by wlad031** [comment]
As I know, it's not necessary

File: templates/layout.html (hunk 5 above)
Comment (dansagunov) [reply templates/layout.html:21]: Found the proof that it is not necessary: https://www.w3.org/TR/html5/grouping-content.html#the-li-element.

**Review by dansagunov — APPROVED** [review]

**Pull request merged**